wcfsetup/install/files/lib/system/endpoint/controller/core/labels/groups/ChangeLabelShowOrder.class.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace wcf\system\endpoint\controller\core\labels\groups;
+
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use wcf\data\label\group\LabelGroup;
+use wcf\data\label\Label;
+use wcf\data\label\LabelEditor;
+use wcf\http\Helper;
+use wcf\system\endpoint\IController;
+use wcf\system\endpoint\PostRequest;
+use wcf\system\label\LabelHandler;
+use wcf\system\showOrder\ShowOrderHandler;
+use wcf\system\showOrder\ShowOrderItem;
+use wcf\system\WCF;
+
+/**
+ * API endpoint for changing the show order of the labels in a label group.
+ *
+ * @author      Marcel Werk
+ * @copyright   2001-2025 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.2
+ */
+#[PostRequest('/core/labels/groups/{id:\d+}/label-show-order')]
+final class ChangeLabelShowOrder implements IController
+{
+    public function __invoke(ServerRequestInterface $request, array $variables): ResponseInterface
+    {
+        WCF::getSession()->checkPermissions(['admin.content.label.canManageLabel']);
+
+        $labelGroup = Helper::fetchObjectFromRequestParameter($variables['id'], LabelGroup::class);
+        $items = \array_map(
+            static fn(Label $label) => new ShowOrderItem($label->labelID, $label->getTitle()),
+            LabelHandler::getInstance()->getLabelGroup($labelGroup->groupID)->getLabels()
+        );
+
+        $sortedItems = (new ShowOrderHandler($items))->getSortedItemsFromRequest($request);
+        $this->saveShowOrder($sortedItems);
+
+        return new JsonResponse([]);
+    }
+
+    private function saveShowOrder(array $items): void
```

Comment:
```suggestion
    /**
     * @param list<ShowOrderItem> $items
     */
    private function saveShowOrder(array $items): void
```

Please always document the type of arrays in both parameters and return values.